### DISCUSSION TITLE
Extra Data being added to 'Average'?

### DISCUSSION THREAD

**Discussion by swalt202** [casact/chainladder-python#500] [Q&A] [answered]
Hello,
I'm having a bit of trouble with weighted averages in ChainLadder. Everything looks good until I start doing calculations, then it seems like there's some extra data messing things up. Link Ratios are fine, but when I apply arithmetic, I get weird results. Any ideas on what might be going on? 

When using the sample triangle ('raa') the averages are correct.

### Defining the triangle
```
raf = cl.Triangle(
    df,
    origin="Year",
    development="Accident_Year",
    columns="Amount",
    cummulative = False
)
##Triangle is correct
```
### Calculating the weighted average
```
cl.Development(average='volume').fit(raf).ldf_
-------------------------------------------
             12 -24    24-36    36-48   ....
Output =    [2.4229 | 1.6061 | 1.3861 | 1.2837 | 1.2240 | 1.1794 | 1.1480 | 1.1258 | 1.1122]
Expected =  [1.42230 | 1.0396 | ... increments to 1.0000]
```

### Custom function to calculate the weighted average
```
weighted = []
remove = 1
for col in df_Triangle.columns[:-1]:
    weighted.append(df_Triangle[col+12].sum() / df_Triangle[col][:-remove].sum())
    remove += 1
weighted = np.array(weighted)
weighted
```

Thank you for any help!

**Comment by kennethshsu** [comment]
Hi @swalt202, would you be able to post your data here?

**Comment by swalt202** [comment, marked as answer]
This issue was resolved after switching to the correct Python and Pandas versions. I was under the impression I had erased this discussion, guess not!

Thank you for the quick response.
